include/nds/arm9/math.h

```diff
@@ -106,8 +109,11 @@ static inline int32_t mulf32(int32_t a, int32_t b)
 /// @param a 20.12 positive value.
 static inline void sqrtf32_asynch(int32_t a)
 {
-    REG_SQRTCNT = SQRT_64;
     REG_SQRT_PARAM = ((uint64_t)(uint32_t)a) << 12;
+    if ((REG_SQRTCNT & 0xF) != SQRT_64)
```

Comment:
According to GBATEK, only bit 0 determines the mode for square roots. Why are you using `0xF` as a mask? shouldn't it just be `BIT(0)`? https://problemkaputt.de/gbatek.htm#dsmaths

For the division it's the lowest two bits. Maybe we need to add new defines to `math.h`:

```diff
 #define DIV_64_64           2
 #define DIV_64_32           1
 #define DIV_32_32           0
+#define DIV_MODE_MASK       3
+
 #define DIV_BUSY            (1 << 15)

 #define SQRT_64             1
 #define SQRT_32             0
+#define SQRT_MODE_MASK      1
+
 #define SQRT_BUSY           (1 << 15)
```

And then use `DIV_MODE_MASK` and `SQRT_MODE_MASK`.

Reply:
I was using 0xF because it was convenient and it works, it's probably not optimal. Your suggestion of checking one bit is probably 1 instruction less (better) because it should generate a tst instruction instead of AND+CMP so in the unlucky case  where we are in the wrong mode the cnt write will happen faster.

We could technically do the same for div, but as I mentioned for div that relies on nobody trying to write garbage to the register.

Reply:
> I was using 0xF because it was convenient and it works because the extra bits are unused, it's probably not optimal. Your suggestion of checking one bit is probably 1 instruction less (better) because it should generate a tst instruction instead of AND+CMP so in the unlucky case where we are in the wrong mode the cnt write will happen faster.
> 
> We could technically do the same for div, but as I mentioned for div that relies on nobody trying to write garbage to the register as (11) would result in wrong behaviour.

Sure, but my proposal is using 3 as mask. It would mean doing something like this:
```
if ((REG_DIVCNT & DIV_MODE_MASK) != DIV_64_32)
...
if ((REG_SQRTCNT & SQRT_MODE_MASK) != SQRT_64)
```

Reply:
I've added these changes now.